### PR TITLE
[bitnami/elasticsearch] Add VIB tests

### PR DESCRIPTION
**Pull request by FraPazGal**
Signed-off-by: FraPazGal <fdepaz@vmware.com>

### Description of the change

The main objective of this PR is to publish our Bitnami Elasticsearch container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD. 

### Applicable issues

NA

Tested in:
- Recent action run: https://github.com/bitnami/containers/actions/runs/4435382646

### CODE REVIEW

File: .vib/elasticsearch/goss/elasticsearch.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+command:
+  # ES7 does not support using `elasticsearch --version` without properly configuring it
+  check-app-version:
+    exec: ls /opt/bitnami/elasticsearch/lib/elasticsearch-core-*.jar | grep {{ .Env.APP_VERSION }}
+    exit-status: 0
+file:
+  # Check x-pack is not included
+  /opt/bitnami/elasticsearch/bin/xpack:
+    exists: false
+  /opt/bitnami/elasticsearch/module/xpack:
```
Comment (joancafom) [inline .vib/elasticsearch/goss/elasticsearch.yaml:10]: Isn't it `x-pack`?
```suggestion
  /opt/bitnami/elasticsearch/bin/x-pack:
    exists: false
  /opt/bitnami/elasticsearch/module/x-pack:
```

File: .vib/elasticsearch/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+binaries:
+  - elasticsearch
+  - gosu
+  - java
+  - yq
+directories:
+  - paths:
```
Comment (joancafom) [inline .vib/elasticsearch/goss/vars.yaml:7]: I think you can also check the permissions for them should be set to 775, as per the following line:

https://github.com/bitnami/containers/blob/53225801ec429d4bae24ea3b2b10458e45267398/bitnami/elasticsearch/8/debian-11/rootfs/opt/bitnami/scripts/elasticsearch/postunpack.sh#L14

File: .vib/elasticsearch/goss/elasticsearch.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+command:
+  # ES7 does not support using `elasticsearch --version` without properly configuring it
+  check-app-version:
+    exec: ls /opt/bitnami/elasticsearch/lib/elasticsearch-core-*.jar | grep {{ .Env.APP_VERSION }}
+    exit-status: 0
+file:
```
Comment (joancafom) [inline .vib/elasticsearch/goss/elasticsearch.yaml:6]: I would additionally check the `log4j2.properties` file, as it is affected by this piece of logic in the postunpack:

https://github.com/bitnami/containers/blob/main/bitnami/elasticsearch/8/debian-11/rootfs/opt/bitnami/scripts/libelasticsearch.sh#L913

**Review by joancafom — COMMENTED** [review]
I just added a small additional comment

File: .vib/elasticsearch/goss/elasticsearch.yaml (hunk 3 above)
Comment (FraPazGal) [reply .vib/elasticsearch/goss/elasticsearch.yaml:6]: Added now, thanks!

**Review by joancafom — APPROVED** [review]
LGTM!

**Pull request merged**